demisto_sdk/commands/common/update_id_set.py

```diff
@@ -93,6 +94,37 @@
 ]
 
 
+def should_skip_item_by_mp(file_path: str, marketplace: str):
+    """
+    Checks if the item given (as path) should be part of the current generated id set.
+     The checks are in this order:
+     1. Check if the given item has the right marketplaces under the 'marketplaces' in the item's file,
+     2. Otherwise, we check if the item is inside a pack that it's pack metadata 'marketplaces' field does not include
+     the current marketplace we are creating this id set for.
+     If there are no 'marketplaces' fields in the the item's file and the metadata, the item is set as xsoar only.
+
+    Args:
+        file_path: path to content item
+        marketplace: the marketplace this current generated id set is for
+
+    Returns: True if should be skipped, else False
+
+    """
+
+    # first check, check field 'marketplaces' in the item's file
+    file_type = Path(file_path).suffix
+    try:
+        item_data = get_file(file_path, file_type)
+        if item_data.get('marketplaces'):
+            if marketplace not in item_data.get('marketplaces'):
+                return True
+    except (ValueError, FileNotFoundError, IsADirectoryError):
```

Comment:
when will you get each exception?

Reply:
First one was for files not having a valid structure, that error is being raised by get_file
second and third i got from random files in content (such as the files under content/TestPlaybooks)
we discussed that folder and we said it should be ignored